File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:

        d = {}
        l=[]
        for i in range(len(nums1)):
            if nums1[i] in d:
                d[nums1[i]] += 1
            else:
                d[nums1[i]] = 1
        for j in range(len(nums2)):
            if nums2[j] in d:
                l.append(nums2[j])
                d.pop(nums2[j])
        return l
        

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int,int> mpp;
        vector<int> l;
        for(int i =0; i<nums1.size(); i++){
            mpp[nums1[i]]++;
        }
        for(int j=0; j<nums2.size(); j++){
            if (mpp.find(nums2[j]) != mpp.end()){
                l.push_back(nums2[j]);
                mpp.erase(nums2[j]);
            }
        }
        return l;
        
    }
};